ChromePhp.php

```diff
@@ -377,7 +382,35 @@ protected function _addRow(array $logs, $backtrace, $type)
 
     protected function _writeHeader($data)
     {
-        header(self::HEADER_NAME . ': ' . $this->_encode($data));
+        $header = self::HEADER_NAME . ': ' . $this->_encode($data);
+        // Most HTTPD servers have a default header line length limit of 8kb, must test to avoid 500 Internal Server Error.
+        if (strlen($header) > self::HTTPD_HEADER_LIMIT) {
+            $data['rows'] = [];
+            $data['rows'][] = [
+                [
+                    'ChromePHP Error: The HTML header will surpass the limit of ' .
+                    $this->_formatSize(self::HTTPD_HEADER_LIMIT) . ' (' . $this->_formatSize(strlen($header)) .
+                    ') - You can increase the HTTPD_HEADER_LIMIT on ChromePHP class, according to your Apache ' .
+                    'LimitRequestFieldsize directive'
+                ], '', self::ERROR
+            ];
+            $header = self::HEADER_NAME . ': ' . $this->_encode($data);
+        }
+        header($header);
+    }
+
+    protected function _formatSize($arg) {
+        if ($arg > 0) {
+            $j = 0;
+            $ext = ["bytes","Kb","Mb","Gb","Tb"];
+            while ($arg >= pow(1024, $j)) ++$j; {
```

Comment:
I don't think the `while` loop usage syntax is good.

We should make this `while` syntax be readable if possible.